landingai/vision_pipeline.py

```diff
@@ -204,6 +210,64 @@ def save_image(self, filename_prefix: str, image_src: str = "") -> "FrameSet":
             c += 1
         return self
 
+    def save_video(
+        self,
+        video_file_path: str,
+        video_fps: Optional[int] = None,
+        video_length_sec: Optional[int] = None,
+        image_src: str = "",
+    ) -> "FrameSet":
+        """Save the FrameSet as an mp4 video file. The following example, shows to use save_video to save a clip from a live RTSP source.
+        ```python
+            video_len_sec=10
+            fps=4
+            img_src = NetworkedCamera(stream_url, fps=fps)
+            frs = FrameSet()
+            for i,frame in enumerate(img_src):
+                if i>=video_len_sec*fps: # Limit capture time
+                    break
+                frs.extend(frame)
+            frs.save_video("sample_images/test.mp4",video_fps=fps)
+        ```
+
+        Parameters
+        ----------
+        video_file_path : str
+            Path and filename with extension of the video file
+        video_fps : Union[int, None], optional
+            Either the frames per second or the video length should be provided to assemble the video. if none of the two are provided, the method will try to set a "reasonable" value.
+        video_length_sec : Union[int, None], optional
+            Either the frames per second or the video length should be provided to assemble the video. if none of the two are provided, the method will try to set a "reasonable" value.
```

Comment:
```suggestion
            The total number of seconds for the output video file. This value will be capped at the number of frames available in this Frameset.
            Either the `video_fps` or `video_length_sec` should be provided to assemble the video. if none of the two are provided, the method will try to set a "reasonable" value.
            If both `video_fps` and `video_length_sec` are provided, `video_length_sec` takes precedence over `video_fps`.
```

Reply:
it'd be beneficial to explain:
1. how the "reasonable" value is calculated, the users may get want to know the behavior to decide how to use it, i.e. wether the user should provide a default value.
2. the behavior when both arguments are provided.